llvm/lib/Transforms/Vectorize/VPlanPatternMatch.h

```diff
@@ -221,8 +221,9 @@ struct Recipe_match {
     if ((!matchRecipeAndOpcode<RecipeTys>(R) && ...))
       return false;
 
-    assert(R->getNumOperands() == std::tuple_size<Ops_t>::value &&
-           "recipe with matched opcode the expected number of operands");
+    assert(
+        R->getNumOperands() == std::tuple_size<Ops_t>::value &&
+        "recipe with matched opcode without the expected number of operands");
```

Comment:
the 2 `without` make it a bit hard to grep, maybe something like that would be clearer?
```suggestion
        "recipe with matched opcode does not have the expected number of operands");
```